app/models/case_contact_topic.rb

```diff
@@ -0,0 +1,42 @@
+class CaseContactTopic < ApplicationRecord
+  CASA_DEFAULT_COURT_TOPICS = Rails.root.join("data", "default_contact_topics.yml")
+  REQUIRED_ATTRS = %w[title details].freeze
+
+  belongs_to :source, polymorphic: true
```

Comment:
I think you should split the topics owned by a CasaOrg and those owned by a CaseContact into separate tables and models.

Polymorphism is a cool, high-level design strategy, but it complicates the design anytime it is used. We have some strong developers who work on the project, but we also have some junior ones who will be effectively forced out of working in this area of the application. Imagine a partial that gets shared by both the Org ones and the CC ones. Then imagine this partial references some method from its source. Then imagine a dev puts the method on just one of the two source models. Then it breaks and they don't know why.

I just don't see the benefits of a polymorphic table here.

Reply:
I already commented this in slack but I should probably put it here too so context is not lost. 

![image](https://github.com/rubyforgood/casa/assets/14540596/dc149b41-23b2-48a8-b01f-f2c1b9840e04)

This is the other model structure that me and firelemons were discussing. Thoughts? I think the ids might be off but arrows show general belongs to ness. 

Amount of rows could also be further reduced with a triple join somewhere.